Formula/e/eatmemory.rb

```diff
@@ -0,0 +1,17 @@
+class Eatmemory < Formula
+  desc "Simple program to allocate memory from the command-line"
+  homepage "https://github.com/julman99/eatmemory"
+  url "https://github.com/julman99/eatmemory/archive/refs/tags/v0.1.8.tar.gz"
+  sha256 "1d2bc0976762a01f554948a3c728bc52cb639aa4d5dbba1421fc3677f042c23f"
+  license "MIT"
+
+  def install
+    system "make", "install", "PREFIX=#{prefix}"
+  end
+
+  test do
+    output = shell_output "#{bin}/eatmemory -?"
+    version_escaped = version.to_s.gsub(/\./, '\.')
+    assert_match %r{^eatmemory #{version_escaped} - https://github.com/julman99/eatmemory\n.*}, output
```

Comment:
We need a test that exercises the some of the functionality of the app. Version checks or usage checks (foo --version or foo --help) are not sufficient, as explained in the formula cookbook.

In most cases, a good test would involve running a simple test case: run #{bin}/foo input.txt.

- Then you can check that the output is as expected (with assert_equal or assert_match on the output of shell_output)
- You can also check that an output file was created, if that is expected: assert_predicate testpath/"output.txt", :exist?

Some advice for specific cases:

- If the formula is a library, compile and run some simple code that links against it. It could be taken from upstream's documentation / source examples.
- If the formula is for a GUI program, try to find some function that runs as command-line only, like a format conversion, reading or displaying a config file, etc.
- If the software cannot function without credentials, a test could be to try to connect with invalid credentials (or without credentials) and confirm that it fails as expected.
- Same if the software requires a virtual machine, docker instance, etc. to be running.

Reply:
So this is a program that runs, uses up the amount of memory specified in the command line and then waits until user presses a key to end. 

So before I invest time on this test, is it Ok if I write a test that launches the program asking to use 100MB of RAM, then it checks that 100MB are indeed being used (not sure if this is even possible), and then sends a keystroke so the program dies?

Reply:
I don't think the checking for usage is needed, but just starting and waiting being tested should give us a reasonable confidence it works for users.

Reply:
Hello! I implemented the proper testing, including memory usage. However the [linux tests keep failing](https://github.com/Homebrew/homebrew-core/actions/runs/6280975838/job/17058839567?pr=143810#step:4:55). 

The curious thing is if I run `brew test eatmemory` in a linux machine it works, even more, if I run the same docker image `homebrew/ubuntu22.04`, point it to my branch and then run `brew test-bot --only-formulae --junit --only-json-tab --skip-dependents --testing-formulae="eatmemory" --added-formulae="eatmemory" --deleted-formulae=""` it works perfectly.

<img width="1340" alt="image" src="https://github.com/Homebrew/homebrew-core/assets/508718/c8ac87fa-08fa-4012-abd8-98d18e44f4e9">

Do you have any idea why could this be failing only on the CI server?

Thanks a lot

Reply:
Maybe use less memory and redirect stderr to stdout so we can see it easily?

Reply:
I found the issue, the `ps` command was being truncated because the columns on the shell were too low (probably 80). I changed the env variable for the process to 500 and it worked. I also added more tests for different memory eat sizes.